specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json

```diff
@@ -4559,6 +4654,181 @@
         }
       }
     },
+    "EndpointModelDeprecationProperties": {
+      "type": "object",
+      "properties": {
+        "fineTune": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "fineTuneOn": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "inference": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "inferenceOn": {
+          "format": "date-time",
+          "type": "string"
+        }
+      }
+    },
+    "EndpointModelProperties": {
+      "type": "object",
+      "properties": {
+        "capabilities": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "deprecation": {
+          "$ref": "#/definitions/EndpointModelDeprecationProperties"
+        },
+        "finetuneCapabilities": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "format": {
+          "type": "string"
+        },
+        "isDefaultVersion": {
```

Comment:
>isDefaultVersion [](http://example.com/codeflow?start=9&length=16)

In place of a boolean can this be converted in an enum so that its values can be more extensible in the future?

Reply:
we are tying to align the schema with here: https://github.com/Azure/azure-rest-api-specs/blob/2e5be0e72597c6fc8d438f20e38087d900c16427/specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2024-06-01-preview/cognitiveservices.json#L5914

so it's unlikely that they will extend it in the future.